imports/plugins/core/orders/client/containers/invoiceContainer.js

```diff
@@ -39,44 +54,622 @@ class InvoiceContainer extends Component {
+      // selected but not all of them, loop through the items array and return a
+      // new array with item ids only, then set the selectedItems array with the itemIds
+      const updateEditedItems = [];
+
+      const itemIds = uniqueItems.map((item) => {
+        // on select all refunded quantity should be all existing items
+        updateEditedItems.push({
+          id: item._id,
+          title: item.title,
+          refundedTotal: item.variants.price * item.quantity,
+          refundedQuantity: item.quantity
+        });
+        return item._id;
+      });
+      this.setState({
+        editedItems: updateEditedItems,
+        selectedItems: itemIds,
+        selectAllItems: true,
+        isUpdating: true
+      });
+    }
+  }
+
+  handleInputChange = (event, value, lineItem) => {
+    let { editedItems } = this.state;
+
+    const isEdited = editedItems.find(item => {
+      return item.id === lineItem._id;
+    });
+
+    const refundedQuantity = lineItem.quantity - value;
+
+    if (isEdited) {
+      editedItems = editedItems.filter(item => item.id !== lineItem._id);
+      isEdited.refundedTotal = lineItem.variants.price * refundedQuantity;
+      isEdited.refundedQuantity = refundedQuantity;
+      if (refundedQuantity !== 0) {
+        editedItems.push(isEdited);
+      }
+    } else {
+      editedItems.push({
+        id: lineItem._id,
+        title: lineItem.title,
+        refundedTotal: lineItem.variants.price * refundedQuantity,
+        refundedQuantity
+      });
+    }
+    this.setState({
+      editedItems,
+      value
+    });
+  }
+
+  /**
+   * Media - find media based on a product/variant
+   * @param  {Object} item object containing a product and variant id
+   * @return {Object|false} An object contianing the media or false
+   */
+  handleDisplayMedia = (item) => {
+    const variantId = item.variants._id;
+    const productId = item.productId;
+
+    const variantImage = Media.findOne({
+      "metadata.variantId": variantId,
+      "metadata.productId": productId
+    });
+
+    if (variantImage) {
+      return variantImage;
+    }
+
+    const defaultImage = Media.findOne({
+      "metadata.productId": productId,
+      "metadata.priority": 0
+    });
+
+    if (defaultImage) {
+      return defaultImage;
+    }
+    return false;
+  }
+
+  getRefundedItemsInfo = () => {
+    const { editedItems } = this.state;
+    return {
+      quantity: editedItems.reduce((acc, item) => acc + item.refundedQuantity, 0),
+      total: editedItems.reduce((acc, item) => acc + item.refundedTotal, 0),
+      items: editedItems
+    };
+  }
+
+  hasRefundingEnabled() {
+    const order = this.state.order;
+    const paymentMethodId = getBillingInfo(order).paymentMethod.paymentPackageId;
+    const paymentMethod = Packages.findOne({ _id: paymentMethodId });
+    const paymentMethodName = paymentMethod.name;
+    const isRefundable = paymentMethod.settings[paymentMethodName].support.includes("Refund");
```

Comment:
We need to figure out an interim solution to this since the Paypal methods don't fit this pattern.